build/dart/internal/gen_executable_call.py

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env python3
+#
+# Copyright 2013 The Flutter Authors. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+"""Generates a shell or batch script to run a command."""
+
+import argparse
+import os
+import string
+
+
+def main():
+  parser = argparse.ArgumentParser(description=__doc__)
+  parser.add_argument('--output', required=True, help='Output file')
+  parser.add_argument('--command', required=True, help='Command to run')
+  parser.add_argument('--cwd', required=False, help='Working directory')
+  parser.add_argument('rest', nargs='*', help='Arguments to pass to the command')
+
+  # Rest of the arguments are passed to the command.
+  args = parser.parse_args()
+
+  out_path = os.path.dirname(args.output)
+  if not os.path.exists(out_path):
+    os.makedirs(out_path)
+
+  script = string.Template(
+      '''#!/bin/sh
+
+set -e
```

Comment:
dumb question because I get tripped up on this constantly. set -e makes sure that the exit code of this script is non zero if the dart test exit code is non zero right? Can you check that failed tests report non-zero exit codes through the shell script too?

Reply:
And command or pipeline that has an error will exit the script automatically.  Dart tests that fail set errno 1.